src/domain/Configuration.ts

```diff
@@ -0,0 +1,26 @@
+import Mapping from './Mapping';
+
+/**
+ * Configuration schema interface.
+ */
+interface Configuration {
+  basePath: string;
+  execution: 'parallel' | 'sequential';
+  sql: string[];
+  connectionResolver: string;
+  hooks: {
+    pre_sync: string[];
+    post_sync: string[];
+  };
+  injectedConfig: {
+    vars: Mapping<string>;
+  };
+  migration: {
+    directory: string;
+    tableName: string;
+    // TODO: Only 'sql' is supported sourceType now. JS will be supported later.
+    sourceType: 'sql' | 'javascript';
```

Comment:
@mesaugat This `javascript` based migration file support is currently missing. We can merge this and add that support soon in the next release. 

All we need to add here, is an ability to import { up, down } functions from the migration files. It won't be that hard, given all the groundwork is already done here.

Reply:
Nice.